src/item.h

```diff
@@ -670,6 +677,45 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      */
     int chip_resistance( bool worst = false ) const;
 
+    /** How much damage has the item sustained? */
+    int damage() const {
+        return damage_;
+    }
+
+    /** Minimum amount of damage to an item (state of maximum repair) */
+    int min_damage() const;
+
+    /** Maximum amount of damage to an item (state before destroyed) */
+    int max_damage() const;
+
+    /**
+     * Apply damage to item constrained by @ref min_damage and @ref max_damage
+     * @param qty maximum amount by which to adjust damage (negative permissible)
+     * @param dmg type of damage which may be passed to @ref on_damage callback
+     * @return whether item should be destroyed
+     */
+    bool mod_damage( int qty, damage_type dt = DT_NULL );
```

Comment:
The `bool` return value also makes sense as it matches the current way of handling item damage in the code: in general functions that damage/"exhaust" an item return `true` when it "runs out".

I'm not sure about the damage type argument, though. What would it be used for?
I assume the `mod_damage` function is supposed to represent damage that already went through item's defenses and thus won't be modified later but applied directly.